endpoints.go

```diff
@@ -33,3 +42,402 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
+    }
+
+    stmt, err := db.Prepare("DELETE FROM " + viper.GetString("db_schema") + ".survey WHERE survey_ref = $1")
+    if err != nil {
+        http.Error(w, "SQL statement not prepared", http.StatusInternalServerError)
+        return
+    }
+
+    defer stmt.Close()
+
+    _, err = stmt.Exec(params["surveyRef"])
+    if err != nil {
+        http.Error(w, "SQL statement error" + err.Error(), http.StatusInternalServerError)
+        return
+    }
+
+    err = tx.Commit()
+    if err != nil {
+        http.Error(w, "Error committing database transaction", http.StatusInternalServerError)
+        return
+    }
+
+    logger.Logger.Info("Successfully deleted survey")
+    w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+    w.WriteHeader(http.StatusNoContent)
+}
+
+//Update survey based on JSON request
+func updateSurveyByRef (w http.ResponseWriter, r *http.Request) {
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    var params = mux.Vars(r)
+
+    body, err := ioutil.ReadAll(r.Body)
+    if err != nil {
+        http.Error(w, "Couldn't read message body", http.StatusInternalServerError)
+        return
+    }
+
+    var survey models.Survey
+
+    err = json.Unmarshal(body, &survey)
+    if err != nil {
+        http.Error(w, "Error unmarshalling JSON", http.StatusBadRequest)
+        return
+    }
+
+    if survey.ShortName == "" && survey.LongName == "" && survey.LegalBasis == "" && survey.SurveyMode == "" {
+        http.Error(w, "No values to update", http.StatusBadRequest)
+        return
+    }
+
+    tx, err := db.Begin()
+    if err != nil {
+        http.Error(w, "Error starting database transaction", http.StatusInternalServerError)
+        return
+    }
+    defer tx.Rollback()
+
+    results := db.QueryRow("SELECT * FROM " + viper.GetString("db_schema") + ".survey WHERE survey_ref = $1" , params["surveyRef"])
+    var tempSurvey models.Survey
+    err = results.Scan(&tempSurvey.ID, &tempSurvey.SurveyRef, &tempSurvey.ShortName, &tempSurvey.LongName, &tempSurvey.LegalBasis, &tempSurvey.SurveyMode)
+    if err != nil {
+        if err == sql.ErrNoRows {
+            http.Error(w, "Survey reference not found", http.StatusNotFound)
+            return
+        }
+        http.Error(w, "Check query failed", http.StatusInternalServerError)
+        return
+    }
+
+    newShortName := "short_name"
+    newLongName := "long_name"
+    newLegalBasis := "legal_basis"
+    newSurveyMode := "survey_mode"
+
+    //If the JSON body filled in a field, update the value in the database, else put the un-updated value into the survey object so that it's part of the returned JSON
+    if survey.ShortName != "" {
+        newShortName = survey.ShortName
+    } else {
+        survey.ShortName = tempSurvey.ShortName
```

Comment:
You've assigned to the wrong variable here:

```suggestion
        newShortName = tempSurvey.ShortName
```

otherwise we'd be putting an empty string into the database in these fields